apps/command-palette-wp-admin/src/set-locale.ts

```diff
@@ -0,0 +1,44 @@
+import { defaultI18n } from '@wordpress/i18n';
+import i18n from 'i18n-calypso';
+
+const DEFAULT_LANGUAGE = 'en';
+const ALWAYS_LOAD_WITH_LOCALE = [ 'zh' ];
+
+const getLanguageCodeFromLocale = ( localeSlug: string ) => {
+	if ( localeSlug.indexOf( '-' ) > -1 ) {
+		return localeSlug.split( '-' )[ 0 ];
+	}
+	return localeSlug;
+};
+
+const loadLanguageFile = async ( languageFileName: string ) => {
+	const url = `https://widgets.wp.com/command-palette/languages/${ languageFileName }-v1.1.json`;
+
+	globalThis.fetch( url ).then( async ( response ) => {
+		if ( response.ok ) {
+			const body = await response.json();
+			if ( body ) {
+				// Work with the calypso i18n `translate` etc, this app and the command-palette package
+				// aren't using this, but it might be used in the future.
+				i18n.setLocale( body );
+				// Work with the wordpress i18n `__` etc
+				defaultI18n.setLocaleData( body, __i18n_text_domain__ );
+			}
+		}
+		Promise.reject( response );
+	} );
+};
+
+export default async ( localeSlug: string ) => {
+	const languageCode = getLanguageCodeFromLocale( localeSlug );
+
+	// Load translation file if it's not English.
+	if ( languageCode !== DEFAULT_LANGUAGE ) {
```

Comment:
Do we need to also check `en-gb` here?

Reply:
No, `getLanguageCodeFromLocale` always returns the language, not the dialect, i.e. it'll always return `en` rather than `en-gb`. The language file loaded will come from that language code, except for `zh` (Chinese).

tbh, I'm not fond of this. I _think_ it was initially done in Odyssey to avoid some problems loading l10n for moment.js but I'm not sure precisely what. Ideally we would load es-mx for example and fallback to es if that's not available.